rust/blockstore/src/arrow/provider.rs

```diff
@@ -269,7 +269,11 @@ impl BlockManager {
     }
 
     pub(super) async fn cached(&self, id: &Uuid) -> bool {
-        self.block_cache.get(id).await.ok().is_some()
+        self.block_cache
```

Comment:
wow! This seems like a regression introduced in https://github.com/chroma-core/chroma/pull/2890/files